### PR TITLE
Optional JSON/HTML reports

### PR DESCRIPTION
**Pull request by gnieto**
Fixes https://github.com/SUPERAndroidAnalyzer/super/issues/80

I've moved some code from mod.rs to a couple of new classes to isolate JSON and HTML report generation. If you think that this is not needed at the moment, I can remove this first commit.

On second commit I've added the new flags (JSON/HTML) and prepared the code to generate only HTML by default.

### CODE REVIEW

File: src/results/mod.rs [hunk 1]
```diff
@@ -230,18 +175,27 @@ impl Results {
                 println!("Results folder created. Time to create the reports.");
             }
 
-            self.generate_json_report(config)?;
+            if config.has_to_generate_json() {
+                let mut json_reporter = Json::new();
+                json_reporter.generate(config, self).unwrap();
```
Comment (gnieto) [inline src/results/mod.rs]: Remove unwrap

File: src/results/mod.rs [hunk 2]
```diff
@@ -230,18 +175,27 @@ impl Results {
...
+                    println!("JSON report generated.");
+                    println!("");
+                }
             }
 
-            self.generate_html_report(config, package)?;
+            if config.has_to_generate_html() {
+                let mut handlebars_reporter = HandlebarsReport::new(config.get_template_path(),
+                                                                    package.as_ref().to_owned())
+                    .unwrap();
```
Comment (gnieto) [inline src/results/mod.rs]: Remove unwrap

File: src/results/mod.rs [hunk 3]
```diff
@@ -230,18 +175,27 @@ impl Results {
...
+                handlebars_reporter.generate(config, self).unwrap();
```
Comment (gnieto) [inline src/results/mod.rs]: Remove unwrap

File: src/results/mod.rs [hunk 4]
```diff
@@ -253,186 +207,14 @@ impl Results {
...
-    }
 }
 
 impl Serialize for Results {
     fn serialize<S>(&self, serializer: &mut S) -> StdResult<(), S::Error>
         where S: Serializer
     {
         let now = Local::now();
-        let mut state = serializer.serialize_struct("Results", 22)?;
+        let mut state = serializer.serialize_struct("Results", 23)?;
```
Comment (gnieto) [inline src/results/mod.rs]: Revert this change

**Comment by Chuky9** [comment]
Hello @gnieto !
First of all I would like to thank you for taking part of this project with your work!! 
I have checked your reviews and you still have some comments to resolve. As soon as you resolve them we will do the merge!!

Once again, thanks for collaborating and we encourage you to keep working on this project!

**Comment by gnieto** [comment]
Hi @Chuky9,

if you refer to my own comments, I think they are all fixed (unwraps removed). Let me know if you want that I do some other change.
Even that, something has been merged on develop and I need to rebase my branch.

Whenever I will have time, I will try to do some more PR. Let me know if you want that I take care of some specific task. If not, I'll randomly pick some issue from the list.

File: src/config.rs [hunk 5]
```diff
@@ -1242,4 +1298,15 @@ mod tests {
...
     }
+
+    /// Test to check the default reports to be generated
+    #[test]
+    fn it_generates_html_but_not_json_by_default() {
+        let mut final_config = Config::default();
+        final_config.html = false;
+        final_config.json = false;
+
+        assert!(final_config.has_to_generate_html() == true);
```
Comment (Razican) [inline src/config.rs]: No need to compare to true or false, simply `assert!(final_config.has_to_generate_html());` or `assert!(!final_config.has_to_generate_json())`.

**Review by Razican — COMMENTED** [review]
I only see that small detail. The rest is great :)

File: src/config.rs (hunk 5 above)
Comment (gnieto) [reply src/config.rs]: changed on: https://github.com/SUPERAndroidAnalyzer/super/pull/93/commits/6e0fef97714013e6a776465346e60a911ec738e7

**Comment by gnieto** [comment]
Just one more question: I'm seeing lots of formatting commits. Should I simply run rustfmt (last version)?

**Comment by Razican** [comment]
Yup, we use the latest version of rustfmt with default configuration. In fact, test should not allow bad formatting to pass.

**Comment by Razican** [comment]
Actually, I just saw that tests do not check code formatting. I'll fix it and push it to develop. Then, please rebase.
